test/integration/node-env/package.json

```diff
@@ -1,6 +1,6 @@
 {
   "scripts": {
-    "TEST": "node ../../../pkg/dist-node/index.bin.js --optimize --node-env FOOBAR"
+    "TEST": "node ../../../pkg/dist-node/index.bin.js --optimize --env NODE_ENV=FOOBAR"
```

Comment:
can we test a few more things here:
- `--env FOO` (inherit from env variable, which you can set by passing to the TEST command:  `FOO=1 node ../../../pkg/dist-node/index.bin.js ...`
- `{"env": ...}` config

Feel free to do them all in this one integration test, and to just add ENV variables as needed to mock-test-package.

Reply:
Done. The new config test various priorities/override/sourcing behaviors